### ISSUE TITLE
[2.44] RuntimeException in script execution ('pos' action)

### ISSUE THREAD

**Issue by tchegito** [tchegito/zildo#136]
This is the same symptom that #45 and #127 but it happens again, in different place.

```
map=prisonext (64x64)
sprites=[Perso=Zildo
Coords:(374.14307, 83.2925 0.0)
info=ZILDO
mvt=TOUCHE pv=0 weapon=[SWORD,0] name=nicolas lasthit=last hit by Perso=gardenord
Coords:(324.19003, 61.559998 0.0)
info=ENEMY
mvt=null pv=4 on prisonext, 375.14307, 93.2925 (80 - bank 0), 691.0, 540.0 (83 - bank 0), 691.0, 540.0 (SHADOW), 319.3775, 57.002777 (102 - bank 0)]
persos=[Perso=Zildo
Coords:(374.14307, 83.2925 0.0)
info=ZILDO
mvt=TOUCHE pv=0 weapon=[SWORD,0] name=nicolas lasthit=last hit by Perso=gardenord
Coords:(324.19003, 61.559998 0.0)
info=ENEMY
mvt=null pv=4 on prisonext]
variables={nettleCount=-1, Carlo=[[MOON, 1], 800, 1], allowedDynamite=yes, Dizzie=[[DYNAMITE, 1], 15, 20], moneyStolen=224.0, BilelIgorVillage=[[EMPTY_BAG, 1], 100, 2], [[BLUEDROP, 1], 15, -1], [[DYNAMITE, 1], 100, 2]}
scripts=2 scripts running {[fromActions[0 on [pos hector (2, 0) , wait 10 , pos hector (-2, 0) , wait 30 , pos hector (1, 0) , wait 5 , pos hector (-1, 0) ]],death[4 on [herospecial 1 , music ZildoMort , fadeOut 5 , remove , filter 3 , fadeIn , sound ZildoDying , script zildo VIDE , angle zildo , wait 5 , angle zildo 1 , wait 5 , angle zildo 2 , wait 5 , angle zildo 3 , wait 5 , angle zildo , wait 5 , angle zildo 1 , wait 5 , angle zildo 2 , wait 5 , angle zildo 3 , wait 5 , angle zildo , wait 3 , angle zildo 1 , wait 4 , angle zildo 2 , perso zildo addSpr=0.0 , script zildo MORT , if hero_princess:1,0, wait 59 , fadeOut , end 1 ]],}
lastdialog=Lugdunia
�Vieille femme�: C'est affreux ce qu'ils ont fait à notre Roi ! Que va-t-il nous arriver à présent ?
�Petula�: Il parait que d'étranges créatures rôdent autour de l'île des pêcheurs.
�Nicolas�: Entendu, je m'en occupe. Ne t'inquiète pas, c'est sans doute une erreur.
�Petula�: Je ne sais pas, mais c'est grave !
�Nicolas�: Hein, comment ? Mais ... pourquoi ?
�Petula�: Ton grand-père ! Ils l'ont emmené en prison !
�Nicolas�: ... Oui, que se passe-t-il ?
�Petula�: Réveille toi nicolas !
�Nicolas�: Allez, il est temps de se coucher. Je suis épuisé.
Château de Lugdunia
�Nicolas�: Entendu mon roi. Je vous laisse.
�Roi Trion�: Reviens me voir au château demain à la première heure, et je te donnerai des instructions précises.
�Roi Trion�: Pour l'instant, rentre chez toi. Avant tout, il me faut réfléchir à un plan avec mon fidèle Hector.
�Nicolas�: Je comprends, mon roi, et je ferai tout ce...
�Roi Trion�: Sans lui, je crains le pire. La nature va s'affoler et les dégats seront imprévisibles. Tu ne peux pas imaginer la gravité de la situation, jeune nicolas !
�Roi Trion�: Peu de gens le savent, mais le sceptre n'est pas qu'un bijou de la famille royale. Il nous permet à tous de vivre paisiblement sur les Terres d'Alembrume.
�Roi Trion�: Je vais demander à Maltus de tout faire pour retrouver ma fille, et je te donne ordre de retrouver le sceptre.
�Roi Trion�: C'est affreux ce que tu m'as raconté, jeune nicolas. Les deux joyaux de notre royaume nous ont été enlevés, et il faut absolument les retrouver.
�Roi Trion�: ...terrible ! Raconte moi tout ce qui s'est passé.
�Roi Trion�: Mais ... c'est ...
�Roi Trion�: Comment ?!?
java.lang.RuntimeException: We need valid 'who' or 'what' attribute in action pos hector (2, 0) 
	at zildo.fwk.script.command.ActionExecutor.render(ActionExecutor.java:164)
	at zildo.fwk.script.command.ScriptExecutor.renderAction(ScriptExecutor.java:240)
	at zildo.fwk.script.command.ScriptExecutor.renderElement(ScriptExecutor.java:262)
	at zildo.fwk.script.command.ScriptExecutor.render(ScriptExecutor.java:138)
	at zildo.server.state.ScriptManagement.render(ScriptManagement.java:135)
	at zildo.server.EngineZildo.renderFrame(EngineZildo.java:204)
	at zildo.client.stage.SinglePlayer.updateGame(SinglePlayer.java:172)
	at zildo.client.Client.mainLoop(Client.java:194)
	at com.alembrum.OpenGLRenderer.onDrawFrame(OpenGLRenderer.java:107)
	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1649)
	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1354)
```

And another occurence, with 'moveTo' action:

```
map=foret (64x64)
sprites=[Perso=Zildo
Coords:(564.99255, 1006.3711 0.0)
info=ZILDO
mvt=VIDE pv=5 weapon=[SWORD,0] name=Elijah lasthit=last hit by Perso=g2
Coords:(572.13873, 20.560238 0.0) ==> (329,360)
info=ENEMY
mvt=VIDE pv=2 on bosquet, 565.99255, 1016.3711 (97 - bank 0), 691.0, 540.0 (83 - bank 0), 603.0, 139.0 (SHADOW), 105.19545, 345.466 (103 - bank 0), 648.0, 406.0 (STONE_HEAVY), Perso=pano
Coords:(552.0, 938.0 0.0)
info=NEUTRAL
mvt=VIDE pv=1, 0.0, 0.0 (80 - bank 0), Perso=new
Coords:(523.0, 661.0 0.0) ==> (22,23)
info=ENEMY
mvt=VIDE pv=2, 523.0, 667.0 (80 - bank 0), 0.0, -12.0 (SHADOW_SMALL), Perso=new
Coords:(735.51904, 842.51904 0.0) ==> (699,818)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 735.51904, 841.51904 (SHADOW), Perso=new
Coords:(280.5897, 674.5904 0.0) ==> (288,678)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 280.5897, 673.5904 (SHADOW), Perso=new
Coords:(483.55087, 416.31537 0.0) ==> (483,409)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 483.55087, 415.31537 (SHADOW), Perso=new
Coords:(484.31995, 233.68 0.0) ==> (503,230)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 484.31995, 232.68 (SHADOW)]
persos=[Perso=Zildo
Coords:(564.99255, 1006.3711 0.0)
info=ZILDO
mvt=VIDE pv=5 weapon=[SWORD,0] name=Elijah lasthit=last hit by Perso=g2
Coords:(572.13873, 20.560238 0.0) ==> (329,360)
info=ENEMY
mvt=VIDE pv=2 on bosquet, Perso=pano
Coords:(552.0, 938.0 0.0)
info=NEUTRAL
mvt=VIDE pv=1, Perso=new
Coords:(523.0, 661.0 0.0) ==> (22,23)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(735.51904, 842.51904 0.0) ==> (699,818)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(280.5897, 674.5904 0.0) ==> (288,678)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(483.55087, 416.31537 0.0) ==> (483,409)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(484.31995, 233.68 0.0) ==> (503,230)
info=ENEMY
mvt=VIDE pv=2]
variables={nettleCount=-1, Carlo=[[MOON, 1], 800, 1], allowedDynamite=yes, Dizzie=[[DYNAMITE, 1], 15, 20], BilelIgorVillage=[[EMPTY_BAG, 1], 100, 2], [[BLUEDROP, 1], 15, -1], [[DYNAMITE, 1], 100, 2]}
scripts=1 scripts running {[@scene@bosquet_free_way[1 on [markQuest trip_polaky 1 , exec bosquet_duo ]],}
lastdialog=Lugdunia Forest
�Maltus�: Remember, always keep your head. You cannot be harmed as long as you stay with me.
�Maltus�: So Elijah, are you proud to be going with me on a mission for King Trion ?
�Maltus�: Well done, boy. Let's go !
Polaky
�Elijah�: I'd rather return to see Maltus.
Lugdunia Forest
�Eric�: Do you want to try to get in the house ?
�Eric�: I've always wondered what's inside, but I'm too scared to set foot near there.
�Eric�: Hi ! Did you see the house on the hill ? It has been abandoned for quite a while !
�Maltus�: There are guards patrolling here. Elijah, defeat them and come back as soon as the way is clear.
�Maltus�: This way.
�Maltus�: Follow me, we are close to Polaky.
�Elijah�: Fortunately, my Sword came in handy. So, where do we go now ?
�Maltus�: Run away, me ? Absolutely not ! I was scouting ahead ... to help accomplish our mission !
�Elijah�: You would know if you hadn't run away !
�Maltus�: Ah Elijah ! Where have you been ?
�Elijah�: Maltus !!!
�Maltus�: What's going on ?
�Maltus�: We can't get passed this log.
�Elijah�: Very well, let's go !
�Maltus�: At last Elijah, there you are. I will remove this block, then follow me. Alright ?
Lugdunia
�Mostapha�: But I don't have anything right now ! Come back later !
java.lang.RuntimeException: We need valid 'who' or 'what' attribute in action moveTo maltus (0, -10) 
	at zildo.fwk.script.command.ActionExecutor.render(ActionExecutor.java:164)
	at zildo.fwk.script.command.ScriptExecutor.renderAction(ScriptExecutor.java:240)
	at zildo.fwk.script.command.ScriptExecutor.renderElement(ScriptExecutor.java:262)
	at zildo.fwk.script.command.ScriptExecutor.render(ScriptExecutor.java:138)
	at zildo.server.state.ScriptManagement.render(ScriptManagement.java:135)
	at zildo.server.EngineZildo.renderFrame(EngineZildo.java:204)
	at zildo.client.stage.SinglePlayer.updateGame(SinglePlayer.java:172)
	at zildo.client.Client.mainLoop(Client.java:194)
	at com.alembrum.OpenGLRenderer.onDrawFrame(OpenGLRenderer.java:107)
	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1562)
	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1262)
```

**Comment by tchegito** [comment]
Once again: (09/02/2018)

```
map=foret (64x64)
sprites=[Perso=Zildo
Coords:(564.99255, 1006.3711 0.0)
info=ZILDO
mvt=VIDE pv=2 weapon=[SWORD,0] name=pat lasthit=last hit by Perso=g2
Coords:(572.24866, 20.629238 0.0) ==> (361,284)
info=ENEMY
mvt=VIDE pv=2 on bosquet, 565.99255, 1016.3711 (80 - bank 0), 691.0, 540.0 (83 - bank 0), 696.0, 314.0 (SHADOW), 201.98767, 298.54053 (103 - bank 0), 648.0, 406.0 (STONE_HEAVY), Perso=pano
Coords:(552.0, 938.0 0.0)
info=NEUTRAL
mvt=VIDE pv=1, 0.0, 0.0 (80 - bank 0), Perso=new
Coords:(523.0, 661.0 0.0) ==> (15,8)
info=ENEMY
mvt=VIDE pv=2, 523.0, 667.0 (80 - bank 0), 0.0, -12.0 (SHADOW_SMALL), Perso=new
Coords:(706.93945, 837.0 0.0) ==> (698,837)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 706.93945, 836.0 (SHADOW), Perso=new
Coords:(279.6399, 672.35986 0.0) ==> (291,657)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 279.6399, 671.35986 (SHADOW), Perso=new
Coords:(529.7506, 424.749 0.0) ==> (564,459)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 529.7506, 423.749 (SHADOW), Perso=new
Coords:(490.0, 288.65002 0.0) ==> (490,301)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 490.0, 287.65002 (SHADOW)]
persos=[Perso=Zildo
Coords:(564.99255, 1006.3711 0.0)
info=ZILDO
mvt=VIDE pv=2 weapon=[SWORD,0] name=pat lasthit=last hit by Perso=g2
Coords:(572.24866, 20.629238 0.0) ==> (361,284)
info=ENEMY
mvt=VIDE pv=2 on bosquet, Perso=pano
Coords:(552.0, 938.0 0.0)
info=NEUTRAL
mvt=VIDE pv=1, Perso=new
Coords:(523.0, 661.0 0.0) ==> (15,8)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(706.93945, 837.0 0.0) ==> (698,837)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(279.6399, 672.35986 0.0) ==> (291,657)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(529.7506, 424.749 0.0) ==> (564,459)
info=ENEMY
mvt=VIDE pv=2, Perso=new
Coords:(490.0, 288.65002 0.0) ==> (490,301)
info=ENEMY
mvt=VIDE pv=2]
variables={BilelIgorVillage=[[EMPTY_BAG, 1], 100, 2], [[BLUEDROP, 1], 15, -1], [[DYNAMITE, 1], 100, 2], Carlo=[[MOON, 1], 800, 1], nettleCount=-1, Dizzie=[[DYNAMITE, 1], 15, 20], allowedDynamite=yes}
scripts=1 scripts running {[@scene@bosquet_free_way[1 on [markQuest trip_polaky 1 , exec bosquet_duo ]],}
lastdialog=Forêt de Lugdunia
�Maltus�: Surtout, garde ton sang froid. Il ne peut rien t'arriver avec moi.
�Maltus�: Alors pat, tu es fier de m'accompagner sur une mission du roi Trion ?
�Maltus�: Très bien petit ! Allons y.
Polaky
�Pat�: Je devrais aller chercher Maltus avant.
Forêt de Lugdunia
�Maltus�: Il semble qu'il y a des gardes qui patrouillent par ici. pat, va les combattre et reviens me chercher une fois que la voie est libre.
�Maltus�: Par ici.
�Maltus�: Suis moi, nous ne sommes plus très loin de la Polaky.
�Pat�: Heureusement que j'avais cette épée. Bon, où allons nous maintenant ?
�Maltus�: Moi, enfui ? Mais pas du tout ! Je partais en éclaireur... afin d'accomplir notre mission !
�Pat�: Tu le saurais si tu ne t'étais pas enfui !
�Maltus�: Ah pat ! Où étais tu passé ?
�Pat�: Maltus !!!
�Maltus�: Qu'est ce que... ?
�Maltus�: On ne peut pas aller par là avec cette souche.
�Pat�: Très bien, allons y !
�Maltus�: Ah enfin pat, tu es là. Je vais enfoncer ce plot et après ça, tu me suis. D'accord ?
Lugdunia
�Grand-pÃ¨r�: Le château est juste au sud du village, à côté de la prison.
�Grand-pÃ¨r�: pat, tu es assez grand maintenant. Prend ce collier, il te sera très utile.
Scierie des frères Lamedur.
�Vieille femme�: Tu cherches la forêt ? C'est juste au nord du village.
java.lang.RuntimeException: We need valid 'who' or 'what' attribute in action moveTo maltus (0, -10) 
	at zildo.fwk.script.command.ActionExecutor.render(ActionExecutor.java:164)
	at zildo.fwk.script.command.ScriptExecutor.renderAction(ScriptExecutor.java:240)
	at zildo.fwk.script.command.ScriptExecutor.renderElement(ScriptExecutor.java:262)
	at zildo.fwk.script.command.ScriptExecutor.render(ScriptExecutor.java:138)
	at zildo.server.state.ScriptManagement.render(ScriptManagement.java:135)
	at zildo.server.EngineZildo.renderFrame(EngineZildo.java:204)
	at zildo.client.stage.SinglePlayer.updateGame(SinglePlayer.java:172)
	at zildo.client.Client.mainLoop(Client.java:194)
	at com.alembrum.OpenGLRenderer.onDrawFrame(OpenGLRenderer.java:107)
	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1531)
	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1248)
```

**Comment by tchegito** [comment]
Fixed in 2.45

Two last cases were the same issue: map scroll on the wrong axis. So hero arrives on the wrong map.

But first one isn't totally understood. Weird circumstance is that 4 south guards aren't present, but hero hasn't any sword to kill them. So where are they ? It only remains the first at the north of the map.

So it's paste to another issue in order to validate this one.

082f966a6d1073525305a3ad1e2d707138c3e9dd
